code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -390,10 +390,21 @@
 									span_notice("You cut \the [src] apart with \the [W]."))
 				deconstruct(TRUE)
 				return
+		if(W.tool_behaviour == TOOL_WRENCH && anchorable) // The locker is open and we're using a wrench
+			if(isinspace() && !anchored)
+				return
+			set_anchored(!anchored)
+			W.play_tool_sound(src, 75)
+			user.visible_message(span_notice("[user] [anchored ? "anchored" : "unanchored"] \the [src] [anchored ? "to" : "from"] the ground."), \
+						balloon_alert(user, "You [anchored ? "anchored" : "unanchored"] \the [src] [anchored ? "to" : "from"] the ground."), \
+						span_hear("You hear a ratchet."))
+			return // This return prevents us from putting the wrench inside the locker when we bolt/unbolt it
 		if (user.combat_mode)
 			return FALSE
 		if(user.transferItemToLoc(W, drop_location())) // so we put in unlit welder too
 			return
+	else if(W.tool_behaviour == TOOL_WRENCH && anchorable) // Bolts are on the inside now, so we make sure that the locker isn't closed
+		balloon_alert(user, "Locker must be open to access bolts!")
```

Comment:
balloon alerts are lowercase and not verbose

Reply:
This was pretty much as short as I could think to make it while still giving all the relevant information. Any suggestions on a snappier line?

Reply:
You could just search the code for balloon_alert and see the precedents (e.g. balloon_alert(user, "close the panel first!")) but I'd just do "must be open first!".